src/methods/function.js

```diff
@@ -508,8 +508,8 @@ export function SetFunctionName(
       name = realm.intrinsics.emptyString;
     } else {
       // c. Else, let name be the concatenation of "[", description, and "]".
-      invariant(description !== null);
-      name = new StringValue(realm, `[${description}]`);
+      invariant(description !== null && description instanceof StringValue);
```

Comment:
Change `null` to `undefined`.
Or just drop that part of the disjunction, as the `instanceof` check is sufficient.